torchvision/prototype/models/resnet.py

```diff
@@ -231,11 +242,28 @@ class WideResNet101_2Weights(Weights):
     )
 
 
+# Maybe we should rename kwargs into model_kwargs to be more explicit? This is not a BC-breaking change I believe.
 def resnet18(weights: Optional[ResNet18Weights] = None, progress: bool = True, **kwargs: Any) -> ResNet:
+    # Considering we're showing this warning literally everywhere, it might be worth writing a helper for it
+    # This would make sure the warning message is consistent, avoid code duplication, etc.
```

Comment:
Yes indeed that's the plan. Just wanted to leave it for last moment until I have a clearer idea of how the implementation works for Quantization, Detection etc. I'll make this change soon.

Reply:
Addressed at PR #4948 by commit 1172e1fefce5b3975ab30a78ad3da1673f89942b